### PR TITLE
UGRID-15: Add SonarCloud config to Docker config

### PR DESCRIPTION
**Pull request by ahmad-el-sayed**

### CODE REVIEW

File: scripts/docker/build_and_test.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+fi
+
+# ========================
+# configure
+# ========================
+print_text_box "Configure build"
+cmake \
+  -S . \
+  -B ${build_dir} \
+  -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
```
Comment (lucacarniato) [inline scripts/docker/build_and_test.sh:23]: A potential enhancement would be to verify that all specified environment variables are set; if any are missing, an error should be raised.

File: scripts/docker/build_and_test.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -e
```
Comment (lucacarniato) [inline scripts/docker/build_and_test.sh:3]: set -eo? To exit on error or unset variable

File: scripts/docker/requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+requests
```
Comment (lucacarniato) [inline scripts/docker/requirements.txt:1]: you might want to set a version, to ensure that we will not be affected by changes in request python library

Comment (ahmad-el-sayed) [reply scripts/docker/requirements.txt:1]: set to 2.32.3

File: scripts/docker/build_and_test.sh (hunk 1 above)
Comment (ahmad-el-sayed) [reply scripts/docker/build_and_test.sh:23]: what can be check is the validity value of the strings for ex build type is one of "release", "debug" or "relwithdebinfo". simulated booleans can be "true" or "false". Will do it in separate issue.

**Review by lucacarniato — APPROVED** [review]

**Pull request merged**